google/cloud/internal/timer_queue.h

```diff
@@ -72,10 +75,70 @@ class TimerQueue : public std::enable_shared_from_this<TimerQueue> {
   TimerQueue(TimerQueue&&) = delete;
   TimerQueue& operator=(TimerQueue&&) = delete;
 
-  /// Adds a timer to the queue.
+  /**
+   * Adds a timer to the queue.
+   *
+   * The future returned by this function is satisfied when either:
+   * - The timer expires, in which case the future is satisfied with a
+   *   successful `StatusOr`.
+   * - The timer queue is shutdown (or was shutdown), in which case the future
+   *   is satisfied with an error status.
+   */
   future<StatusOr<std::chrono::system_clock::time_point>> Schedule(
       std::chrono::system_clock::time_point tp);
 
+  /**
+   * Adds a timer to the queue and atomically attach a callback to the timer.
```

Comment:
"atomically" probably deserves a little explanation.  Perhaps just phrase it in terms of who might make the callback?

Reply:
Done.